staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -31,32 +31,37 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/kms"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
 )
 
 const (
 	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
 	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
 	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
+	kmsTransformerPrefixV1       = "k8s:enc:kms:v1:"
 )
 
-// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file
-func GetTransformerOverrides(filepath string) (map[schema.GroupResource]value.Transformer, error) {
+type kmsServiceGetter func(name string, kmsConfig map[string]interface{}) (kms.Service, error)
+
+// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file.
+// It takes an optional kmsServiceGetter as an argument, which is used when a KMS based encryption backend is used.
+func GetTransformerOverrides(filepath string, initKMSService kmsServiceGetter) (map[schema.GroupResource]value.Transformer, error) {
 	f, err := os.Open(filepath)
 	if err != nil {
 		return nil, fmt.Errorf("error opening encryption provider configuration file %q: %v", filepath, err)
 	}
 	defer f.Close()
 
-	result, err := ParseEncryptionConfiguration(f)
+	result, err := ParseEncryptionConfiguration(f, initKMSService)
```

Comment:
It seems like the configuration is very provider/KMS specific. Rather than trying to do empty interface back flips to pass the config and its parse method around, we should bind the config in prior to passing it in here.